week1/intro.cpp

```diff
@@ -0,0 +1,27 @@
+class Foo {
+public:
+    std::mutex two,three;
+    Foo() {
+        two.lock();
+        three.lock();    
+    }
+
+    void first(function<void()> printFirst) {
+        // printFirst() outputs "first". Do not change or remove this line.
+        printFirst();
+        two.unlock();
```

Comment:
I think `lock` and `unlock` must be called from the same thread for `mutex`: https://en.cppreference.com/w/cpp/thread/mutex/unlock#:~:text=Unlocks%20the%20mutex.,ownership%20of%20the%20same%20mutex.